src/sidebar/services/test/api-test.js

```diff
@@ -17,395 +16,283 @@ const util = require('../../../shared/test/util');
-      proxyquire(
-        '../api',
-        util.noCallThru({
-          angular: angular,
-          '../retry-util': {
-            retryPromiseOperation: function(fn) {
-              return fn();
-            },
-          },
-        })
-      )
-    );
-  });
-
-  beforeEach(function() {
-    sandbox = sinon.sandbox.create();
+  function defaultBodyForStatus(status) {
+    if (status === 204) {
+      return '';
+    } else if (status >= 500) {
+      return '<html><body>Internal Server Error</body></html>';
+    } else {
+      return {};
+    }
+  }
+
+  /**
+   * Expect an HTTP call using `window.fetch`.
+   *
+   * @param {string} method - Expected HTTP method (lower case)
+   * @param {string} pathAndQuery - Expected part of URL after API root
+   * @param {number} status -
+   *   Expected HTTP status. If <= 0 then the call to `fetch` will reject with
+   *   the content of `body` as the error message.
+   * @param {Object|string} body - Expected response body
+   */
+  function expectCall(
+    method,
+    pathAndQuery,
+    status = 200,
+    body = defaultBodyForStatus(status)
+  ) {
+    const url = `https://example.com/api/${pathAndQuery}`;
+    if (status > 0) {
+      fetchMock.mock(url, { status, body }, { method });
+    } else {
+      fetchMock.mock(
+        url,
+        {
+          status,
+          throws: new Error(body),
+        },
+        { method }
+      );
+    }
+  }
 
+  beforeEach(() => {
     const fakeApiRoutes = {
       links: sinon.stub(),
-      routes: sinon.stub(),
+      routes: sinon.stub().returns(Promise.resolve(routes)),
     };
     fakeAuth = {
-      tokenGetter: sinon.stub(),
+      tokenGetter: sinon.stub().returns(Promise.resolve('faketoken')),
     };
     fakeStore = {
       apiRequestStarted: sinon.stub(),
       apiRequestFinished: sinon.stub(),
     };
 
-    angular.mock.module('h', {
-      apiRoutes: fakeApiRoutes,
-      auth: fakeAuth,
-      settings: { apiUrl: 'https://example.com/api/' },
-      store: fakeStore,
-    });
-
-    angular.mock.inject(function(_$q_) {
-      $q = _$q_;
-      fakeAuth.tokenGetter.returns($q.resolve('faketoken'));
+    api = apiFactory(fakeApiRoutes, fakeAuth, fakeStore);
 
-      fakeApiRoutes.routes.returns($q.resolve(routes));
+    fetchMock.catch(() => {
+      throw new Error('Unexpected `fetch` call');
     });
   });
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
-    sandbox.restore();
+  afterEach(() => {
+    fetchMock.restore();
   });
 
-  beforeEach(
-    angular.mock.inject(function(_$httpBackend_, _api_) {
-      $httpBackend = _$httpBackend_;
-      api = _api_;
-    })
-  );
-
-  it('saves a new annotation', function(done) {
-    api.annotation.create({}, {}).then(function(saved) {
-      assert.isNotNull(saved.id);
-      done();
-    });
+  it('saves a new annotation', () => {
+    expectCall('post', 'annotations', 201, { id: 'new-id' });
```

Comment:
Sadly, our API returns a 200 for this. In looking at this I realized no functest existed for same, so [I've just added one](https://github.com/hypothesis/h/pull/5576).

I notice a later test specifies 200, so maybe I'm misreading what this test _does_

Reply:
You read the test right. I'll change this to a 200.